mempool/clist_mempool.go

```diff
@@ -434,6 +441,9 @@ func (mem *CListMempool) resCbFirstTime(tx types.Tx, res *abci.CheckTxResponse)
 		tx:        tx,
 	}) {
 		mem.notifyTxsAvailable()
+		if mem.onNewTx != nil {
+			go mem.onNewTx(tx) // non-blocking to not slow down txs processing
```

Comment:
There is no backpressure mechanism here.
In our latest [QA tests](https://github.com/cometbft/cometbft/blob/v1.x/docs/references/qa/CometBFT-QA-v1.md#blocks-produced-per-minute-and-transactions-processed-per-minute) in version `v1.x`, with 200 nodes, 175 of them being validators, CometBFT is able to sustain a workload of around 25000 (1k-long) transactions per minute.
We are concerned that allowing the number or goroutines to grow here with no further mechanism to limit goroutine creation might:

* negatively affect the results we get on [memory used](https://github.com/cometbft/cometbft/blob/v1.x/docs/references/qa/CometBFT-QA-v1.md#memory-resident-set-size) by CometBFT in future versions
* or even worse, that the CometBFT node might run out of memory and crash

Reply:
You're right, do you have any suggestion?